### DISCUSSION TITLE
Component of water in PorousFlow Kernels

### DISCUSSION THREAD

**Discussion by navier-rgb** [idaholab/moose#26671] [Q&A Modules: Porous Flow] [answered]
Hello, It’s probably trivial, but I want to be sure. If I’m using the PorousFlow kernels, for example, `PorousFlowMassTimeDerivative` and `PorousFlowDispersiveFlux`, what is the assignment order of the fluid components, assuming I’m using three components: water, a, b? Is water the last component or the first (component 0)? Thanks.

**Comment by GiudGiud** [comment]
Hello

The ordering of the components is set by the ordering of the variables in the PorousFlowDictator user object. Do you have one in your simulation?
For example in that input
```
[UserObjects]
  [dictator]
    type = PorousFlowDictator
    porous_flow_vars = 'pp sat'
    number_fluid_phases = 2
    number_fluid_components = 2
  []
```

Guillaume

**Reply by navier-rgb** [reply]
I see. So, will I have the following, assuming I’m using the `PorousFlowMassTimeDerivative` Kernel?

```
 [kernels] 
  [mass_Na]
    type = PorousFlowMassTimeDerivative
    fluid_component = 0
    variable = sat
  [] 
  [mass_Na]
    type = PorousFlowMassTimeDerivative
    fluid_component = 1
    variable = pp
  []
[] 
```
Or variable pp will have `fluid_component = 0 `instead?

**Reply by GiudGiud** [reply]
you need to set the fluid component always. 
I m not sure why it has a default @cpgr

**Reply by cpgr** [reply, marked as answer]
The PDE's are written as fluid component mass balances (so one equation for each fluid component, summed over all fluid phases, see https://mooseframework.inl.gov/modules/porous_flow/governing_equations.html). The index you assign each component just has to be consistent between the kernels and the mass fraction material (usually PorousFlowMassFraction), so that it uses the correct fraction of mass for each component.

The default fluid component (and fluid phase) index is 0, as the most basic model is one fluid in one phase, so I guess it was done to make the inout files for this basic case as simple as possible.

**Reply by navier-rgb** [reply]
thanks @cpgr @GiudGiud
